### DISCUSSION TITLE
Does composer or pyquafu has the controlled Ry gate？

### DISCUSSION THREAD

**Discussion by OS-Batman** [ScQ-Cloud/pyquafu#66] [General]
I want to implement operations like controlled Ry gate or controlled X gate in the circuit，but i didn't find。
If there is such an operation, can someone tell me how to achieve it。

**Comment by Zhaoyilunnn** [comment]
controlled X gate is equivalent to CNOT gate (Use either `cnot` or `cx` in `QuantumCircuit`). Controlled Ry gate is not supported yet. Maybe you could build circuit using other frameworks and transform to pyquafu circuit for execution on quafu platform.

**Reply by OS-Batman** [reply]
thanks for your advice^^

**Reply by chensgit169** [reply]
Sorry for just seeing this message. Actually ``pyquafu`` supports auto-decomposition of unitary, you may try ``QuantumCircuit.unitary(matrix, pos)``. And there are ``CXMatrix`` and  ``ry_mat`` in ``quafu.elements.element_gates.matrices``, by which it should be straightforward to build a matrix of ``C-ry``.
